src/main/java/com/google/cloud/flink/bigquery/ProtobufUtils.java

```diff
@@ -0,0 +1,717 @@
+ * ProtoType.
+ */
+@Internal
+public final class ProtobufUtils {
+  private static final String MAPTYPE_ERROR_MESSAGE = "MapType is unsupported.";
+  private static final int MAX_BIGQUERY_NESTED_DEPTH = 15;
+  private static final String EMPTY_STRING = "";
+  private static final String RESERVED_NESTED_TYPE_NAME = "STRUCT";
+
+  private static final ImmutableMap<Field.Mode, DescriptorProtos.FieldDescriptorProto.Label>
+      BigQueryModeToProtoFieldLabel =
+          new ImmutableMap.Builder<Field.Mode, DescriptorProtos.FieldDescriptorProto.Label>()
+              .put(Field.Mode.NULLABLE, DescriptorProtos.FieldDescriptorProto.Label.LABEL_OPTIONAL)
+              .put(Field.Mode.REPEATED, DescriptorProtos.FieldDescriptorProto.Label.LABEL_REPEATED)
+              .put(Field.Mode.REQUIRED, DescriptorProtos.FieldDescriptorProto.Label.LABEL_REQUIRED)
+              .build();
+
+  private static final ImmutableMap<LegacySQLTypeName, DescriptorProtos.FieldDescriptorProto.Type>
+      BigQueryToProtoType =
+          new ImmutableMap.Builder<LegacySQLTypeName, DescriptorProtos.FieldDescriptorProto.Type>()
+              .put(LegacySQLTypeName.BYTES, DescriptorProtos.FieldDescriptorProto.Type.TYPE_BYTES)
+              .put(LegacySQLTypeName.INTEGER, DescriptorProtos.FieldDescriptorProto.Type.TYPE_INT64)
+              .put(LegacySQLTypeName.BOOLEAN, DescriptorProtos.FieldDescriptorProto.Type.TYPE_BOOL)
+              .put(LegacySQLTypeName.FLOAT, DescriptorProtos.FieldDescriptorProto.Type.TYPE_DOUBLE)
+              .put(
+                  LegacySQLTypeName.NUMERIC, DescriptorProtos.FieldDescriptorProto.Type.TYPE_STRING)
+              .put(
+                  LegacySQLTypeName.BIGNUMERIC,
+                  DescriptorProtos.FieldDescriptorProto.Type.TYPE_INT64)
+              .put(LegacySQLTypeName.STRING, DescriptorProtos.FieldDescriptorProto.Type.TYPE_STRING)
+              .put(
+                  LegacySQLTypeName.TIMESTAMP,
+                  DescriptorProtos.FieldDescriptorProto.Type.TYPE_STRING)
+              .put(LegacySQLTypeName.DATE, DescriptorProtos.FieldDescriptorProto.Type.TYPE_INT64)
+              .put(
+                  LegacySQLTypeName.DATETIME,
+                  DescriptorProtos.FieldDescriptorProto.Type.TYPE_STRING)
+              .put(
+                  LegacySQLTypeName.GEOGRAPHY,
+                  DescriptorProtos.FieldDescriptorProto.Type.TYPE_BYTES)
+              .put(LegacySQLTypeName.TIME, DescriptorProtos.FieldDescriptorProto.Type.TYPE_STRING)
+              .put(LegacySQLTypeName.RECORD, DescriptorProtos.FieldDescriptorProto.Type.TYPE_STRING)
+              .build();
+
+  public static FlinkFnApi.Schema.FieldType toProtoType(LogicalType logicalType) {
+    return logicalType.accept(new ProtobufUtils.LogicalTypeToProtoTypeConverter());
+  }
+
+  public static ProtoSchema toProtoSchema(Schema schema) throws IllegalArgumentException {
```

Comment:
Do you require all 3 `toProtoSchema` methods? Or will just one suffice? I ask because in PR #21 I only see the RowType schema being called

Reply:
@VinayLondhe14 ,  These are run time access methods, some times input to this method is RowType, Schema and StructuredType.